### ISSUE TITLE
Implement basic message fetching

### ISSUE THREAD

**Issue by dasch** [zendesk/ruby-kafka#47]
Fetching messages from a single partition should be easy to implement.

**Comment by mensfeld** [comment]
Would be awesome if there would be an option for autocommit after data is being fetched. Also isn't there a way to get offset automatically from Kafka/Zookeeper to keep reading after restart or on a next fetch loop?

**Comment by dasch** [comment]
Offset commits only make sense if there's the notion of a consumer group involved. This would just be for ad-hoc fetches, although some of the plumbing could be re-used when group consumers need to fetch messages.

**Comment by mensfeld** [comment]
@dasch  yeah but consumer groups will be introduced right? ;)

**Comment by dasch** [comment]
Sure – but they'll be separate from this. I imagine this will look something like:

```ruby
kafka = Kafka.new(...)

messages = kafka.fetch_messages(topic: "impressions", partition: 42, offset: 1234)
```

So there'll be no coordination. That's fine for some, non-distributed workflows, and should provide us with feature parity for Poseidon.

In theory it's possible to build a simplistic 0.8 compatible consumer on top of this, but I'm going to go for the full 0.9 API instead.

**Comment by mensfeld** [comment]
@dasch basic means basic and indeed for non-distributed workflows it will act great but having groups in 1.0 is in my opinion a must be.

**Comment by dasch** [comment]
Yup, that's on the roadmap for 1.0. I just need to make sure the Producer is rock solid before embarking on such a task.

**Comment by mrkamel** [comment]
+1

**Comment by dasch** [comment]
Implemented in #51.
